src/Util/Input.cpp

```diff
@@ -25,12 +25,17 @@ bool Input::IsMouseMoving() const {
 bool Input::IfScroll() const {
     return m_Scroll;
 }
-std::pair<int, int> Input::GetScrollDistance() const {
+glm::vec2 Input::GetScrollDistance() const {
     return m_ScrollDistance;
 }
 void Input::Update() {
-    SDL_GetMouseState(&m_CursorPosition.first, &m_CursorPosition.second);
+    std::array<int, 2> temp;
+    SDL_GetMouseState(&temp[0], &temp[1]);
+    m_CursorPosition.x = static_cast<float>(temp[0]);
+    m_CursorPosition.y = static_cast<float>(temp[1]);
```

Comment:
should it be like this because x and y are not "really" related values
```suggestion
    int x, y;
    SDL_GetMouseState(&x, &y);
    m_CursorPosition.x = static_cast<float>(x);
    m_CursorPosition.y = static_cast<float>(y);
```